src/ethereum/osaka/bal_utils.py

```diff
@@ -0,0 +1,338 @@
+"""
+BAL Utilities for EIP-7928
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Utilities for working with Block Access Lists, including hashing and validation.
+"""
+
+from typing import Union, Optional
+from ethereum_types.bytes import Bytes
+from ethereum_types.numeric import Uint
+
+from ethereum.crypto.hash import Hash32, keccak256
+
+from .ssz_types import (
+    BlockAccessList,
+    AccountChanges,
+    SlotChanges,
+    SlotRead,
+    StorageChange,
+    BalanceChange,
+    NonceChange,
+    CodeChange,
+    MAX_TXS,
+    MAX_SLOTS,
+    MAX_ACCOUNTS,
+    MAX_CODE_SIZE,
+)
+
+
+def compute_bal_hash(bal: BlockAccessList) -> Hash32:
+    """
+    Compute the hash of a Block Access List.
+    
+    The BAL is SSZ-encoded and then hashed with keccak256.
+    
+    Parameters
+    ----------
+    bal :
+        The Block Access List to hash.
+        
+    Returns
+    -------
+    hash :
+        The keccak256 hash of the SSZ-encoded BAL.
+    """
+    bal_bytes = ssz_encode_block_access_list(bal)
+    return keccak256(bal_bytes)
+
+
+def ssz_encode_uint(value: Union[int, Uint], size: int) -> bytes:
```

Comment:
The ssz functions should probably live in an `ssz` module, at the very least so we can have a docstring explaining what SSZ stands for 🤣